dot/sync/chain_sync.go

```diff
@@ -843,17 +843,41 @@ func (cs *chainSync) processBlockData(blockData types.BlockData, origin blockOri
 	announceImportedBlock := cs.getSyncMode() == tip
 
 	if blockData.Header != nil {
+		var setFinalisedHash func() error = nil
+		if blockData.Justification != nil && len(*blockData.Justification) > 0 {
```

Comment:
nit: We can move this condition to a `hasJustification()` method in `types.BlockData`